### PR TITLE
Add activity record creation to contribution services

### LINKED ISSUE #336: API: Add activity record to Contribution Model
Add an activity record creation to updateContribution and createContribution services.

For update, it should read:

UserName updated #ConrtibutionID fields. fieldName changed from "value" to "value"

For created, it should read:

UserName created #ConrtibutionID.

For archived, it should read

UserName archived #ConrtibutionID.

### PR DESCRIPTION
**Pull request by iankerriganharris**
Closes #336

### CODE REVIEW

File: api/models/entity/Activity.ts [hunk 1]
```diff
@@ -5,27 +5,29 @@ import {
...
-import {Expenditure} from "./Expenditure";
+import { Expenditure } from './Expenditure';
 
 export enum ActivityTypeEnum {
     USER = 'user',
     PERMISSION = 'permission',
     CAMPAIGN = 'campaign',
     GOVERNMENT = 'government',
     INVITATION_EMAIL = 'invitation email',
+    CONTRIBUTION = 'contribution'
```
Comment (iankerriganharris) [inline api/models/entity/Activity.ts]: @danmelton for create/update/archive contribution events, should the activity type be 'contribution' or 'user'?

File: api/services/contributionService.ts [hunk 2]
```diff
@@ -85,7 +90,16 @@ export async function addContributionAsync(contributionAttrs: IAddContributionAt
...
             if (await contribution.isValidAsync()) {
-                return await contributionRepository.save(contribution);
+                const saved = await contributionRepository.save(contribution);
+                await createActivityRecordAsync({
+                    currentUser: user,
+                    notes: `${user.email} created ${saved.id}.`,
+                    campaign: contribution.campaign,
+                    government: contribution.government,
+                    activityType: ActivityTypeEnum.CONTRIBUTION,
+                    activityId: user.id
```
Comment (danmelton) [inline api/services/contributionService.ts]: this should be the contribution id

File: api/services/contributionService.ts [hunk 3]
```diff
@@ -178,7 +193,21 @@ export async function updateContributionAsync(contributionAttrs: IUpdateContribu
...
+                    .map(k => `${k} changed from ${contribution[k]} to ${attrs[k]}.`)
+                    .join(' ')
+            ]);
+            await createActivityRecordAsync({
+                currentUser: user,
+                notes: `${user.email} updated ${contributionAttrs.id} fields. ${changeNotes}`,
+                campaign: contribution.campaign,
+                government: contribution.government,
+                activityType: ActivityTypeEnum.CONTRIBUTION,
+                activityId: user.id
```
Comment (danmelton) [inline api/services/contributionService.ts]: contribution id here

File: api/services/contributionService.ts [hunk 4]
```diff
@@ -229,16 +258,28 @@ export async function archiveContributionAsync(contrAttrs: IArchiveContributionB
...
         if (hasCampaignPermissions && contribution.status === ContributionStatus.DRAFT) {
             contribution.status = ContributionStatus.ARCHIVED;
             await contributionRepository.save(contribution);
+            await createActivityRecordAsync({
+                currentUser: user,
+                notes: `${user.email} archived ${contribution.id}.`,
+                campaign: contribution.campaign,
+                government: contribution.government,
+                activityType: ActivityTypeEnum.CONTRIBUTION,
+                activityId: user.id
```
Comment (danmelton) [inline api/services/contributionService.ts]: contribution id

**Pull request merged**